### PR TITLE
RN-973 Upgrade Meditrak

### PR DESCRIPTION
**Pull request by IgorNadj**
## Package updates:

Start from a brand new react-native. Add up to date packages and get it building. Both Android and iOS.

**DEPRECATED PACKAGES LIST**

- react-native-camera
  - react-native-vision-camera
  - must upgrade, old version causing all sorts of issues
  - ✅ Upgraded

- react-native-qrcode-scanner
  - The recommendation of most developers that were previously using this library is to consider using react-native-vision-camera with its plugin vision-camera-code-scanner
  - ✅ Upgraded

- react-native-side-menu
  - Recommendation is react-native-side-menu-updated
  - 🟡 Not upgraded, works fine as is, will leave for a future PR

- react-navigation
  - Recommendation is @react-navigation/native
  - 🟡 Not upgraded, works fine as is, will leave for a future PR
  - Take react-navigation-redux-helpers with it?
    - Yes, not needed, only used to persist state, can be done with core lib: https://reactnavigation.org/docs/state-persistence/

- sha256 
  - 🟡 Not upgraded, works fine as is, will leave for a future PR

## AppCenter changes:

Fixed the builds for Android and iOS. 
iOS has been re-setup to build and release through AppCenter following AppCenter documentation.

## Manual release tasks
- [ ] Update Slab for Meditrak app releasing

### CODE REVIEW

File: packages/meditrak-app/README.md [hunk 1]
```diff
@@ -1,176 +1,88 @@
-# @tupaia/meditrak
+# Prerequisites
 
-Android and iOS app for surveying medical facilities
+- nvm use
```
Comment (rohan-bes) [inline packages/meditrak-app/README.md:3]: I had to add an `.nvmrc` file with `16.20.2` in it to the root of the `meditrak-app` package

File: packages/meditrak-app/package.json [hunk 2]
```diff
@@ -1,107 +1,85 @@
...
     "react-navigation-redux-helpers": "^4.0.1",
-    "react-navigation-stack": "^2.8.2",
-    "react-qr-code": "^2.0.8",
-    "react-redux": "^5.0.7",
-    "realm": "^6.1.5",
-    "redux": "^3.7.2",
-    "redux-persist": "^5.9.1",
+    "react-navigation-stack": "^2.10.4",
+    "react-redux": "^8.1.2",
+    "realm": "^12.1.0",
```
Comment (rohan-bes) [inline packages/meditrak-app/package.json:53]: When testing this locally I was having issues with a missing `realm-constants.json` file in `node_modules/ream`. There seems be something that some other users have encountered: https://github.com/realm/realm-js/issues/6270

Apparently there's meant to be a post-install script that creates that file but something seemed to be going wrong for me.

Bumping to `12.3.0` fixed the issue though, currently `yarn.lock` has resolved to `12.2.1`

**Review by rohan-bes — COMMENTED** [review]
Thanks for this @IgorNadj!

Ran through a local install and had a minor issue with `realm` but aside from that all working well. Wanna go through this together on Friday and then hopefully we can merge :+1:

File: packages/meditrak-app/package.json (hunk 2 above)
Comment (kneth) [reply packages/meditrak-app/package.json:53]: Try to upgrade to v12.3.1 as we have a fix for it :smile:

**Review by rohan-bes — APPROVED** [review]
Awesome stuff, thanks for all the work!

File: packages/meditrak-app/package.json (hunk 2 above)
Comment (rohan-bes) [reply packages/meditrak-app/package.json]: Thanks @kneth :pray:

**Comment by IgorNadj** [comment]
Merging before test pass because this only affects MediTrak App, which is on a change freeze pending this upgrade.

I have done a smoke test on the build and run of web-frontend and it passed, any other issues will be caught by release regression testing.

**Pull request merged**